lenskit-core/src/main/java/org/lenskit/data/entities/AbstractEntity.java

```diff
@@ -57,6 +57,16 @@ public EntityType getType() {
     /**
      * {@inheritDoc}
      *
+     * This implementation delegates to {@link #hasAttribute(String)}
+     */
+    @Override
+    public boolean hasAttribute(TypedName<?> name) {
+        return hasAttribute(name.getName());
```

Comment:
Its not clear to me that this is correct. Wouldn't you want to check name and type?